src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.TryGet.cs

```diff
@@ -125,7 +125,7 @@ public bool TryGetSingleValue(out float value)
             }
 
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            char standardFormat = span.IndexOfAny((byte)'e', (byte)'E') >= 0 ? 'e' : default;
+            char standardFormat = _numberHasExponent ? 'e' : default;
```

Comment:
Is it useful to store the char `standardFormat` as a field (something like `_numerParsingFormat`) that can be passed directly to `Utf8Parser` rather than the bool `_numberHasExponent` to avoid having this check?

It would end up growing the struct by a bit more (`sizeof(char) > sizeof(bool)`) but maybe its worth it.

Reply:
God point, it doesn't change much (no measurable diff outside of error margins) in terms of performance (in neither direction) but it cleaner and more readable IMO. See 90816a8